performance.lic

```diff
@@ -52,37 +63,64 @@ class Performance
       when /muster the strength to perform/
         pause 30
       when /You begin (.*)/, /You continue (.*)/, /You find yourself relaxing (.*)/, /You concentrate on your sense of rhythm (.*)/, /You cannot use the/
-        exit if done_training?
+        if done_training?
+          wear_hand_armor_and_put_away_instrument
+          exit
+        end
       when /You finish (.*)/
         if done_training?
+          wear_hand_armor_and_put_away_instrument
           exit
         else
-          play_song?(@settings, @song_list, is_instrument_worn, skip_clean)
+          DRC.play_song?(@settings, song_list, is_instrument_worn, skip_clean)
         end
       when /Perhaps you should find somewhere drier/
-        exit
+        if done_training?
+          wear_hand_armor_and_put_away_instrument
+          exit
+        end
       end
     end
   end
 
+  def remove_hand_armor_and_get_instrument
+    if DRCI.remove_item?(@hand_armor)
+      @hand_armor_removed = true
+      DRCI.stow_item?(@hand_armor)
+    end
+    if DRCI.get_item?(@instrument)
+      @instrument_out = true
+    end
+  end
+
+  def wear_hand_armor_and_put_away_instrument
+    if @instrument_out
+      DRCI.stow_item?(@instrument)
+    end
+    if @hand_armor_removed
+      DRCI.get_item?(@hand_armor)
+      DRCI.wear_item?(@hand_armor)
+    end
+  end
+
   def done_training?
-    DRSkill.getxp('Performance') > @settings.performance_mindstate_goal
+    if @mindstate_goal
+      DRSkill.getxp('Performance') >= @mindstate_goal
+    else
+      DRSkill.getxp('Performance') >= @settings.performance_mindstate_goal
+    end
   end
 
   def stop_play
-    return unless @did_play
-    return if @no_instrument
-    @did_play = false
     DRC.stop_playing
     Flags['ct-song'] = true
   end
 end
 
 before_dying do
   fput('stop play')
-
   instrument = get_settings.instrument
-  DRC.bput("stow #{instrument}", 'You put') if instrument
+  DRCI.stow_item?(instrument) if instrument && DRCI.in_hands?(instrument)
```

Comment:
Is this still necessary now that we're cleaning up in `wear_hand_armor_and_put_away_instrument`?

Reply:
For the case when you kill performance manually, this makes sense, no?

Reply:
I'm not sure. I mean, most scripts aren't designed to be killed at any arbitrary point in their execution (or at least, it's assumed they'll leave things in an unknown state). Most scripts don't try to clean up anything beyond Flags if they're killed. 

Should we re-wear our hand armor too? We don't have access to the instance var (I think), so it'd be a matter of either wearing it no matter what or checking if we're already wearing it and then trying to wear it?